product-service/src/createProduct.js

```diff
@@ -0,0 +1,59 @@
+import response from "./utils/response";
+import isEmpty from "validator/es/lib/isEmpty";
+import isInt from "validator/es/lib/isInt";
+import { endClient, getClient, query, queryOne } from "./utils/db";
+import HttpError from "./utils/httpError";
+
+const validateData = (title, price, count) => {
+    const messages = {};
+    if (!title || isEmpty(title)) {
+        messages.title = 'Invalid title';
+    }
+
+    if (!price || !isInt(price, { min: 0 })) {
+        messages.price = 'Invalid price';
+    }
+
+    if (!count || !isInt(count, { min: 0 })) {
+        messages.count = 'Invalid count';
+    }
+
+    if (messages.length) {
+        throw new HttpError(422, messages.join('.'));
```

Comment:
Maybe it's better to use 400 status code?

Reply:
I think 422 status is more appropriate in this case. The difference with 400 is that request syntax is correct, but request body validation fails.

The 422 (Unprocessable Entity) status code means the server understands the content type of the request entity (hence a 415(Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions.